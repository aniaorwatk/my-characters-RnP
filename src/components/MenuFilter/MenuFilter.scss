@import "../../scss/Main";

.menuFilter{
    width: 100%;
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
   margin: 15px 0; 
}

@media only screen and (min-width: $breakpointL) {
    .menuFilter{
        margin-left: 15px;
        justify-content: space-between;
    }
  }