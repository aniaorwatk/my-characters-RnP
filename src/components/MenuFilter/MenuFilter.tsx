import "./MenuFilter.scss";

interface IMenuFilterType {
    searchBox: JSX.Element,
    btnBox: JSX.Element
}

const MenuFilter = ({ searchBox, btnBox }: IMenuFilterType) => {
    return (
        <div className="menuFilter">
            {searchBox}
            <div className="menuFilter__btnBox">{btnBox}</div>
        </div>
    )

}

export default MenuFilter
