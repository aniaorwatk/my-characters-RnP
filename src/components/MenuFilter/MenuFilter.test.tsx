import { render, screen } from "@testing-library/react";
import MenuFilter from "./MenuFilter";

test("renders MenuFilter with search box", () => {
  render(
    <MenuFilter searchBox={<div>search box</div>} btnBox={<div>xxxxx</div>} />
  );

  const searchBox = screen.getByText("search box");

  expect(searchBox).toBeInTheDocument();
});

test("renders MenuFilter with button", () => {
  render(
    <MenuFilter
      searchBox={<div>search box</div>}
      btnBox={<button>button</button>}
    />
  );

  const searchBox = screen.getByText("button");

  expect(searchBox).toBeInTheDocument();
});

test("renders snapshot", () => {
  const view = render(
    <MenuFilter searchBox={<div>xxxxx</div>} btnBox={<div>xxxxx</div>} />
  );

  expect(view.baseElement).toMatchSnapshot();
});
