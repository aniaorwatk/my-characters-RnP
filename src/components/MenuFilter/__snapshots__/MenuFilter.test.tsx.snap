// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<body>
  <div>
    <div
      class="menuFilter"
    >
      <div>
        xxxxx
      </div>
      <div
        class="menuFilter__btnBox"
      >
        <div>
          xxxxx
        </div>
      </div>
    </div>
  </div>
</body>
`;
